pkg/kubelet/kubelet.go

```diff
@@ -2508,15 +2505,16 @@ func (kl *Kubelet) GetKubeletContainerLogs(podFullName, containerName string, lo
 		podStatus = pod.Status
 	}
 
-	if err := kl.validatePodPhase(&podStatus); err != nil {
-		// No log is available if pod is not in a "known" phase (e.g. Unknown).
-		return fmt.Errorf("Pod %q in namespace %q : %v", name, namespace, err)
+	switch podStatus.Phase {
+	case api.PodRunning, api.PodSucceeded, api.PodFailed:
+	default:
+		return fmt.Errorf("pod %q is in an unknown state and logs are not available", pod.Name)
```

Comment:
After looking at this some more, I'm going to remove this check.  If the pod can reference the container, and the container exists, we should show logs.  There's no point to excessively gating which logs we can return.